service/controller/app/resource/chart/desired.go

```diff
@@ -85,28 +89,42 @@ func generateAnnotations(input map[string]string) map[string]string {
 	return annotations
 }
 
-func generateConfig(cr v1alpha1.App, appCatalog v1alpha1.AppCatalog, chartNamespace string) v1alpha1.ChartSpecConfig {
+func (r *Resource) generateConfig(ctx context.Context, cr v1alpha1.App, appCatalog v1alpha1.AppCatalog, chartNamespace string) (v1alpha1.ChartSpecConfig, error) {
 	config := v1alpha1.ChartSpecConfig{}
 
 	if hasConfigMap(cr, appCatalog) {
+		configMapName := key.ChartConfigMapName(cr)
+		cm, err := r.k8sClient.CoreV1().ConfigMaps(chartNamespace).Get(ctx, configMapName, metav1.GetOptions{})
```

Comment:
The CM is in the tenant cluster so I think this needs to check for `tenant.IsAPINotAvailable(err)`. Also if the configmap doesn't exist I think we should just not set the resource version. 

Same for getting the secret. WDYT?

Reply:
Good call, I will extend the unittest for both cases.

Reply:
I found two points we can discuss further. 

> The CM is in the tenant cluster so I think this needs to check for tenant.IsAPINotAvailable(err)

the operator checks this in `current.go` and we cancel the resource when the tenant cluster is unavailable. 
https://github.com/giantswarm/app-operator/blob/e89b6ff14e49ed4834b3734b9adb183e7c098423/service/controller/app/resource/chart/current.go#L65
So we don't need the same error checking code for `generateConfig` function. 

> Also if the configmap doesn't exist I think we should just not set the resource version.

Actually I think the app-operator has to leave the config as blank when it could not find the relevant resource. Otherwise, chart-operator would try to fetch that resource on their own reconciliation. @rossf7 WDYT?

Reply:
> the operator checks this in current.go and we cancel the resource when the tenant cluster is unavailable.

OK cool I just wanted to be sure we handle the TC being unavailable. But like you said since this is done in current.go I think this is fine. 👍 

> Actually I think the app-operator has to leave the config as blank when it could not find the relevant resource.

Yes we just need to set the resource version to speed things up. If it doesn't exist I think the behavior should be the same as now.